MAINT: Common library migration (#254)
ansys/pyaedt-toolkits-antenna: CONTRIBUTING.md

# Contributing

> We absolutely welcome any code contributions and we hope that this
> guide will facilitate an understanding of the PyAEDT antenna toolkit code
> repository. It is important to note that while the PyAEDT antenna toolkit
> software package is maintained by ANSYS and any submissions will be reviewed
> thoroughly before merging, we still seek to foster a community that can
> support user questions and develop new features to make this software
> a useful tool for all users. As such, we welcome and encourage any
> questions or submissions to this repository.
> 
> For contributing to this project, please refer to the [PyAnsys Developer's Guide].
> 
> [PyAnsys Developer's Guide]: https://dev.docs.pyansys.com/index.html

Reviewer: ```suggestion
We welcome code contributions and hope that this
page explains how you can contribute to the AEDT Antenna Toolkit. While this
Python package is maintained by Ansys and any submissions are reviewed
thoroughly before merging, we still seek to foster a community that can
support user questions and develop new features to make this package
a useful tool for all users. As such, we welcome and encourage any
questions or submissions to this repository.

For overall information on contributing to this project, see the [PyAnsys developer's guide].

[PyAnsys Developer's Guide]: https://dev.docs.pyansys.com/index.html
```

> ## Directing Issues and Features Requests
> 
> For reporting bugs and proposing new features, please use the Issues tab instead of the Discussions tab. This will help us track and prioritize work in a more organized manner.

Reviewer: ```suggestion
## Issues and features requests

To report bugs and propose new features, use the **Issues** page instead of the **Discussions** page. This helps us track and prioritize work in a more organized manner.
```

> ## Purpose of Discussions
> 
> The Discussions tab should primarily be used for general questions and discussions about the project. This could include discussions about potential improvements, the future direction of the project, brainstorming ideas, help with using the software, and other topics that don't exactly fit as an Issue.
> 
> Remember, maintaining distinct places for different types of interactions helps keep our project organized and moving forward in a coordinated manner.

Reviewer: ```suggestion
## Discussions

To post general questions and discussions about the project, use the **Discussions** page. Posts can include discussions about potential improvements, questions about the future direction of the project, brainstorming of ideas, help with using the package, and other topics that don't exactly fit as an issue.

Remember, maintaining distinct places for different types of interactions helps us to keep our project organized and move forward in a coordinated manner.
```

## Modify the Code

Everyone can contribute to this project, irrespective of their level of expertise. Your diverse skills, perspectives, and experiences are valuable and we welcome them.

> Here's a simple overview of how you can start making contributions:
> 
> **Fork the Repository:** "Forking" means creating a personal copy of this repository on your GitHub account.
> 
> **Clone the Repository:** After forking, you need to download the repository to your local machine. This can be done using the `git clone` command.
> 
> **Create a New Branch:** A branch is used to isolate your changes from the main project. You can create a new branch using the `git branch` command. Remember to switch to your new branch with the `git checkout` command.
> 
> **Commit Your Changes:** After making your changes, you need to "commit" them. A commit is a packaged set of alterations. Use `git add` to add your files to staging, and then `git commit -m "your message"` to commit them.
> 
> **Push Your Changes:** After committing your changes, "push" them to your forked repository on GitHub with `git push origin <branch-name>`.
> 
> **Create a Pull Request:** A Pull Request (PR) lets us know you have changes you think should be included in the main project. Go to your forked repository on GitHub and click on the 'Pull request' button.
> 
> Following these steps ensures that your contributions will be easily reviewed and potentially included in the project much faster.
> 
> Please don't get discouraged if you find these steps complex, we are here to help you throughout the process.
> 
> We hope these rules will make the Discussions section a better place for every contributor.

Reviewer: ```suggestion
Here's an overview of how you can start making contributions:

**Fork the repository:** Forking means creating a personal copy of this repository on your GitHub account.

**Clone the repository:** After forking, you must download the repository to your local machine. This can be done using the ``git clone`` command.

**Create a branch:** A branch is used to isolate your changes from the main project. You can create a branch using the `git branch` command. Remember to switch to your new branch with the `git checkout` command.

**Commit Your Changes:** After making your changes, you need to "commit" them. A commit is a packaged set of alterations. Use `git add` to add your files to staging, and then `git commit -m "your message"` to commit them.

**Push Your Changes:** After committing your changes, "push" them to your forked repository on GitHub with `git push origin <branch-name>`.

**Create a Pull Request:** A Pull Request (PR) lets us know you have changes you think should be included in the main project. Go to your forked repository on GitHub and click on the 'Pull request' button.

Following these steps ensures that your contributions will be easily reviewed and potentially included in the project much faster.

Please don't get discouraged if you find these steps complex, we are here to help you throughout the process.

We hope these rules will make the Discussions section a better place for every contributor.
```
